var/spack/repos/builtin/packages/r-rstan/package.py

```diff
@@ -18,14 +18,17 @@ class RRstan(RPackage):
     the need to derive the partial derivatives."""
 
     homepage = "http://mc-stan.org/"
-    url      = "https://cran.r-project.org/src/contrib/rstan_2.10.1.tar.gz"
-    list_url = "https://cran.r-project.org/src/contrib/Archive/rstan"
+    url      = "https://cloud.r-project.org/src/contrib/rstan_2.10.1.tar.gz"
+    list_url = "https://cloud.r-project.org/src/contrib/Archive/rstan"
 
+    version('2.19.2', sha256='31e4ceb9c327cd62873225097ffa538c2ac4cb0547c52271e52e4c7652d508da')
     version('2.18.2', sha256='4d75dad95610d5a1d1c89a4ddbaf4326462e4ffe0ad28aed2129f2d9292e70ff')
     version('2.17.2', '60f4a0284c58f5efc1b1cbf488d7edda')
     version('2.10.1', 'f5d212f6f8551bdb91fe713d05d4052a')
 
-    depends_on('r-ggplot2', type=('build', 'run'))
+    depends_on('r@3.0.2:', when='@:2.17.3', type=('build', 'run'))
+    depends_on('r@3.4.0:', when='@2.18.1:', type=('build', 'run'))
+    depends_on('r-ggplot2@2.0.0:', type=('build', 'run'))
     depends_on('r-stanheaders', type=('build', 'run'))
     depends_on('r-inline', type=('build', 'run'))
     depends_on('r-gridextra', type=('build', 'run'))
```

Comment:
```suggestion
    depends_on('r-gridextra@2.0.0:', type=('build', 'run'))
```

Reply:
Also, Rcpp (≥ 0.12.0), BH (≥ 1.69.0)

Reply:
Those, and some other changes were made. There were duplicate dependency specifications and unneeded when conditions for some.